WordPress/gutenberg
# Experimental: try new "Widget Group" block to act as container for Widgets

## Description
Context https://github.com/WordPress/gutenberg/issues/32723.

This approach tries adding a new block called "Widget Box". It's basically just a container to add:

1. A Widget title.
2. InnerBlocks (your Widget).

This mimics the original widget markup format.

It also comes with some additional helpful touches:

1. A template for the inner blocks to pre-add the heading block for the user.
2. A transform to allow users to automatically convert a single widget to be wrapped in a Widget Box.

~Note that this could complement https://github.com/WordPress/gutenberg/pull/33878 where we provide a block pattern~. We can just[ use a `transform`](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-transforms/) which is what I've done now.

Closes https://github.com/WordPress/gutenberg/issues/32723

## To to

- [x] Add a new block that wraps a "title" and any of the supported widget area inner blocks
- [x] Decide wether we want to support nested widget boxes
- [ ] Render the block serverside and account for `before_title`, `after_title` for the rendered title markup
- [ ] Iterate on the UX of the block so that the differences between the editor and the front end are not a surprise


## How has this been tested?

* Go to Widgets screen.
* Add Widget Box block.
* FIll out the heading.
* Add a Widget block of your choice.
* Repeat.
* Test on front end.

To compare against the original/legacy widget markup pattern:

* Download and activate "Classic Widgets" Plugin.
* Go to previous/old widgets screen.
* Add a Widget and give it a title.
* Disable Plugin.
* Return to block-based Widgets screen.
* Add a Widget that is the same as the one you added in the Classic Widgets.
* Click on the "Transform" menu in the block toolbar.
* Click "Widget Box" to transform your block to be wrapped in a Widget Box.
* Fill in title.
* Save/Publish.
* View frontend of site.
* Check markup is near enough identical between legacy and block based widget (when using Widget Box).

## Screenshots 

Tested using TwentyFifteen. The aim is visual parity when trying to create an "Archives" Widget using the "Widget Box" block 

<img width="646" alt="Screen Shot 2021-08-05 at 09 56 37" src="https://user-images.githubusercontent.com/444434/128322368-46d34b69-eec9-4621-9654-7c1be725aef3.png">


https://user-images.githubusercontent.com/444434/128347001-b7cd3e18-2175-42c7-a75e-4f25d798f7b2.mp4

## Types of changes





## Checklist:
